pkg/render/common/elasticsearch/service.go

```diff
@@ -25,13 +25,30 @@ const (
 	linseedEndpoint     = "https://tigera-linseed.%s.svc"
 	httpsFQDNEndpoint   = "https://tigera-secure-es-gateway-http.%s.svc.%s:9200"
 	linseedFQDNEndpoint = "https://tigera-linseed.%s.svc.%s"
+
+	// Handle for multi tenant
+	guardianEndpoint       = "https://guardian.calico-system.svc"
+	guardianFQDNEndpoint   = "https://guardian.calico-system.svc.%s"
+	fluentdLinseedEndpoint = "https://tigera-linseed"
 )
 
-func LinseedEndpoint(osType rmeta.OSType, clusterDomain, namespace string) string {
-	if osType == rmeta.OSTypeWindows {
+func LinseedEndpoint(osType rmeta.OSType, clusterDomain, namespace string, isManagedCluster bool, isFluentd bool) string {
+	switch {
+	// In a managed cluster, all Elasticsearch requests to Linseed are redirected via the Guardian service.
+	// Clients using the Linseed client are automatically configured with the correct SNI for certificate validation.
+	// Since Fluentd doesn't use the Linseed client, we expose an external service named "tigera-linseed" that redirects to Guardian.
+	// The Linseed certificate is already configured to accept connections with SNI set to "tigera-linseed".
+	case isManagedCluster && isFluentd:
+		return fluentdLinseedEndpoint
+	case isManagedCluster && osType == rmeta.OSTypeWindows:
+		return fmt.Sprintf(guardianFQDNEndpoint, clusterDomain)
```

Comment:
Why do we include cluster domain in the endpoint? Is there a situation where this adds a working configuration that would not be possible without? Could we omit `clusterDomain` from all endpoints in `service.go`?
cc: @caseydavenport 

I'm just finding the config a bit confusing, especially in the example where mgmt and mgd have a different cluster domain set.

Reply:
@rene-dekker it's because Windows clients require the FQDN for DNS resolution in some capacity (I don't recall the specifics) so the host header is always set to the FQDN when the client is a fluentd pod on a Windows node.